framework/src/org/checkerframework/framework/type/BoundsInitializer.java

```diff
@@ -133,6 +111,31 @@ public static void initializeTypeArgs(AnnotatedDeclaredType declaredType) {
         declaredType.typeArgs = Collections.unmodifiableList(typeArgs);
     }
 
+    /**
+     * Returns a wildcard whose upper bound is the same as {@code typeVariable}. If the upper bound
+     * is an intersection, then this method returns an unbound wildcard.
+     */
+    private static WildcardType getUpperBoundAsWildcard(
+            TypeVariable typeVariable, AnnotatedTypeFactory factory) {
+        TypeMirror upperBound = typeVariable.getUpperBound();
+        switch (upperBound.getKind()) {
+            case ARRAY:
+            case DECLARED:
+            case TYPEVAR:
+                return factory.types.getWildcardType(upperBound, null);
+            case INTERSECTION:
+                // Can't create a wildcard with an intersection as the upper bound, so use
+                // an unbound wildcard instead.  The extends bound of the
+                // AnnotatedWildcardType be initialized properly elsewhere.
```

Comment:
I'm confused by this comment that the upper bound is properly initialized elsewhere.
Why do we need the code in this location at all then? Why not always return an unbounded wildcard, if logic to initialize the bound already is elsewhere?

Reply:
The upper bound of the AnnotatedWildcardType is initialized else where, but this method is returning the type mirror that will be used as the underlying type of the AnnotatedWildcardType.  It's best if upper bound of underlying type matches the upper bound of the AnnotatedWildcardType.

Reply:
It would be useful to make that mismatch explicit: that here you create an unbound wildcard as the underlying type, which will have a different bound than the AnnotatedWildcardType. I had read the comment as saying that the bound for the unbound wilcard will eventually be set later.
Can you reference where the AWT bound is correctly set?
Would it make sense to also set the bound of the underlying type when the bound of the AWT is changed?